scenario/auto_map_pointer_source.yml

```diff
@@ -0,0 +1,44 @@
+input:
+    input.go: |
+        package example
+
+        // goverter:converter
+        type Converter interface {
+            // goverter:autoMap Address
+            Convert(Person) FlatPerson
+        }
+
+        type FlatPerson struct {
+            Name    string
+            Age     int
+            Street  string
+        }
+        type Person struct {
+            Name    string
+            Age     int
+            Address *Address
+        }
+        type Address struct {
+            Street  string
+        }
+success: |
+    // Code generated by github.com/jmattheis/goverter, DO NOT EDIT.
+
+    package generated
+
+    import execution "github.com/jmattheis/goverter/execution"
+
+    type ConverterImpl struct{}
+
+    func (c *ConverterImpl) Convert(source execution.Person) execution.FlatPerson {
+    	var exampleFlatPerson execution.FlatPerson
+    	exampleFlatPerson.Name = source.Name
+    	exampleFlatPerson.Age = source.Age
+    	var pString *string
+    	if source.Address != nil {
+    		pString = &source.Address.Street
+    	}
+    	xstring := *pString
```

Comment:
This code will produce a
```
panic: runtime error: invalid memory address or nil pointer dereference
```
goverter must never panic, the code could be written like this to prevent this.
```go
    func (c *ConverterImpl) Convert(source execution.Person) execution.FlatPerson {
    	var exampleFlatPerson execution.FlatPerson
    	exampleFlatPerson.Name = source.Name
    	exampleFlatPerson.Age = source.Age
    	var xstring string
    	if source.Address != nil {
    		xstring = (*source.Address).Street
    	}
    	exampleFlatPerson.Street = xstring
    	return exampleFlatPerson
    }
```

Reply:
Agree.

Reply:
Should be fixed.

Reply:
I found one more problem with null pointers. Please don't merge this right now.

Reply:
Fixed in https://github.com/jmattheis/goverter/pull/65/commits/74698915f955025e99b39964d65b56b3ef43a42a.

Now implementation of if-block is looking really same to the `(*Pointer) Build` by the code layout.